SalsaCompiler/SalsaCompiler/main.swift

```diff
@@ -98,12 +106,14 @@ document.pages.enumerated().forEach { index, page in
     guard let pageJsonString = page.simplified().toSketchJson().toJsonString() else { fatalError("Failed to parse page \(index + 1)") }
     try pageJsonString.write(to: url, atomically: false, encoding: String.Encoding.utf8)
   } catch {
+    cleanupWorkingDirectory()
     fatalError("Failed to save page \(index + 1)")
   }
 }
 
 // Create document.json
-guard let documentJson = document.toSketchJson(pageIds: document.pages.reversed().flatMap { IdentifierStore.identifier(forPageNamed: $0.name) }).toJsonString() else {
+guard let documentJson = document.toSketchJson(pageIds: document.pages.reversed().compactMap { IdentifierStore.identifier(forPageNamed: $0.name) }).toJsonString() else {
```

Comment:
Ooh this includes a Swift 4.1 update, are there any configs that need to be changed with this? I'm surprised that there isn't even like an xcodeproj field that wants to be increased...

Reply:
i think xcode 9.3 automatically starts using 4.1 and theres no actual change that needed to be made to turn it on

Reply:
Sweet, I'm used to seeing `LastSwiftUpdateCheck` or `LastUpgradeCheck` changes in xcodeprojs and just wanted to double check.